varats/varats/plots/bug_fixing_relations.py

```diff
@@ -0,0 +1,340 @@
+import logging
+import typing as tp
+from datetime import datetime
+from pathlib import Path
+
+import numpy as np
+import plotly.graph_objs as gob
+import pygit2
+
+from varats.plot.plot import Plot, PlotDataEmpty
+from varats.project.project_util import (
+    get_project_cls_by_name,
+    get_local_project_git,
+)
+from varats.provider.bug.bug import RawBug
+from varats.provider.bug.bug_provider import BugProvider
+
+LOG = logging.getLogger(__name__)
+
+
+def _plot_chord_diagram_for_raw_bugs(
+    project_name: str, bug_set: tp.FrozenSet[RawBug]
+) -> gob.FigureWidget:
+    """Creates a chord diagram representing relations between introducing/fixing
+    commits for a given set of RawBugs."""
+    project_repo = get_local_project_git(project_name)
+
+    # maps commit hex -> node id
+    map_commit_to_id: tp.Dict[str, int] = _map_commits_to_nodes(project_repo)
+    commit_count = len(map_commit_to_id.keys())
+    commit_type: tp.Dict[str, str] = {}
+
+    for commit in project_repo.walk(
+        project_repo.head.target.hex, pygit2.GIT_SORT_TIME
+    ):
+        commit_type[commit.hex] = 'default'
+
+    # if less than 2 commits, no graph can be drawn!
+    if commit_count < 2:
+        raise PlotDataEmpty
+
+    commit_coordinates = _compute_node_placement(commit_count)
+
+    commit_distance_thresholds = [
+        0,
+        round(0.25 * commit_count),
+        round(0.5 * commit_count),
+        round(0.75 * commit_count), commit_count
+    ]
+
+    def _get_commit_interval(distance: float) -> int:
+        """Get right interval for given commit distance using distance
+        thresholds, interval indices are in [0,3] for 5 thresholds."""
+        k = 0
+        while commit_distance_thresholds[k] < distance:
+            k += 1
+        return k - 1
+
+    edge_colors = ['#d4daff', '#84a9dd', '#5588c8', '#6d8acf']
+
+    node_colors = {
+        'fix': 'rgba(0, 177, 106, 1)',
+        'introduction': 'rgba(240, 52, 52, 1)',
+        'introducing fix': 'rgba(235, 149, 50, 1)',
+        'head': 'rgba(142, 68, 173, 1)',
+        'fixing head': 'rgba(142, 68, 173, 1)',
+        'default': 'rgba(232, 236, 241, 1)'
+    }
+
+    # draw relations and preprocess commit types
+    nodes = []
+    lines = []
+    for bug in bug_set:
+        bug_fix = bug.fixing_commit
+        fix_ind = map_commit_to_id[bug_fix]
```

Comment:
```suggestion
        fix_id = map_commit_to_id[bug_fix]
```
Looks like a typo? Did you mean `fix_id`?